Compiler error: suggest braces around initialization of subobject

These are actually only warnings but cause the build to fail (OpenBSD 6.2 -current, clang 4.0.0) due to -Werror.

    Building: src/xhtml/xhtml_node.cpp
    src/xhtml/xhtml_node.cpp:783:38: warning: suggest braces around initialization of subobject [-Wmissing-braces]
                                            css::Specificity specificity = {9999, 9999, 9999};
                                                                            ^~~~~~~~~~~~~~~~
                                                                            {               }
    1 warning generated.
    
    Building: src/xhtml/xhtml_border_info.cpp
    src/xhtml/xhtml_border_info.cpp:315:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
                    std::array<FixedPoint, 4> border = { dims.border_.top, dims.border_.left, dims.border_.bottom, dims.border_.right };
                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                         {                                                                           }
    1 warning generated.
    
    Building: src/xhtml/xhtml_background_info.cpp
    src/xhtml/xhtml_background_info.cpp:182:42: warning: suggest braces around initialization of subobject [-Wmissing-braces]
                            std::array<glm::vec2, 4> quadrant = { glm::vec2(-1.0f, -1.0f), glm::vec2(1.0f, -1.0f), glm::vec2(1.0f, 1.0f), glm::vec2(-1.0f, 1.0f) };
                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                  {                                                                                             }
    1 warning generated.

--- a/src/xhtml/xhtml_background_info.cpp
+++ b/src/xhtml/xhtml_background_info.cpp
@@ -179,7 +179,7 @@ namespace xhtml
 			corners[2] = glm::vec2(right, bottom);
 			corners[3] = glm::vec2(left, bottom);
 
-			std::array<glm::vec2, 4> quadrant = { glm::vec2(-1.0f, -1.0f), glm::vec2(1.0f, -1.0f), glm::vec2(1.0f, 1.0f), glm::vec2(-1.0f, 1.0f) };
+			std::array<glm::vec2, 4> quadrant = {{ glm::vec2(-1.0f, -1.0f), glm::vec2(1.0f, -1.0f), glm::vec2(1.0f, 1.0f), glm::vec2(-1.0f, 1.0f) }};
 
 			const float width = right - left;
 			const float height = bottom - top;

--- a/src/xhtml/xhtml_border_info.cpp
+++ b/src/xhtml/xhtml_border_info.cpp
@@ -312,7 +312,7 @@ namespace xhtml
 		if(styles_ == nullptr) {
 			return;
 		}
-		std::array<FixedPoint, 4> border = { dims.border_.top, dims.border_.left, dims.border_.bottom, dims.border_.right };
+		std::array<FixedPoint, 4> border = {{ dims.border_.top, dims.border_.left, dims.border_.bottom, dims.border_.right }};
 
 		auto& outset = styles_->getBorderImageOutset();
 		for(auto side = 0; side != 4; ++side) {

--- a/src/xhtml/xhtml_node.cpp
+++ b/src/xhtml/xhtml_node.cpp
@@ -780,7 +780,7 @@ namespace xhtml
 				auto attr = n->getAttribute("style");
 				if(attr) {
 					auto plist = css::Parser::parseDeclarationList(attr->getValue());
-					css::Specificity specificity = {9999, 9999, 9999};
+					css::Specificity specificity = {{9999, 9999, 9999}};
 					n->mergeProperties(specificity, plist);
 				}
 			}

